token-lending/program/src/processor.rs

```diff
@@ -1714,6 +1724,69 @@ fn unpack_mint(data: &[u8]) -> Result<Mint, LendingError> {
     Mint::unpack(data).map_err(|_| LendingError::InvalidTokenMint)
 }
 
+fn get_pyth_product_attribute<'a, T>(ite: &mut T) -> String
+where
+    T: Iterator<Item = &'a u8>,
+{
+    let mut len = *ite.next().unwrap() as usize;
+    let mut val = String::with_capacity(len);
+    while len > 0 {
+        val.push(*ite.next().unwrap() as char);
+        len -= 1;
+    }
+    val
+}
+
+fn get_pyth_price(
+    pyth_price_account_info: &AccountInfo,
+    clock: &Clock,
+) -> Result<u64, ProgramError> {
+    let pyth_price_data = &pyth_price_account_info.try_borrow_data()?;
+    let pyth_price = pyth_client::cast::<pyth_client::Price>(pyth_price_data);
+
+    match pyth_price.ptype {
+        pyth_client::PriceType::Price => {}
+        _ => {
+            msg!("Oracle price type is invalid");
+            return Err(LendingError::InvalidOracleConfig.into());
+        }
+    }
+
+    if pyth_price.valid_slot < clock.slot - SLOTS_PER_MINUTE {
+        msg!("Oracle price is stale");
+        return Err(LendingError::InvalidOracleConfig.into());
+    }
+
+    let price: u64 = pyth_price
+        .agg
+        .price
+        .checked_abs()
```

Comment:
wow, prices can be negative?? that seems bad, can we get pyth to change that

Reply:
They have varying `PriceType` for price objects. I thought maybe they are used differently, and a negative value for one might indicate a change since last or something. I'll definitely try to ask.

Reply:
Clarified that's what this is for:

> Price used is i64 because prices can be sometimes negative 
> By price, it can range across multiple price types ; price of a future or yield can be negative
> Also a publisher might publish -1 in price as a null price so it also covers this situation

Reply:
Makes sense, thanks for checking